src/targets/gpu/fuse_mlir.cpp

```diff
@@ -133,12 +138,57 @@ struct find_conv_pointwise
         std::copy_if(ins->inputs().begin(),
                      ins->inputs().end(),
                      std::back_inserter(inputs),
-                     [&](auto input) { return input != conv_ins; });
-        inputs.insert(inputs.end(), conv_ins->inputs().begin(), conv_ins->inputs().end());
+                     [&](auto input) { return input != gemm_based_op; });
+        inputs.insert(inputs.end(), gemm_based_op->inputs().begin(), gemm_based_op->inputs().end());
         mpm.get_module().replace_instruction(
-            ins, mlir_conv{conv_ins->get_operator()}, inputs, {mm});
+            ins, mlir_anchor_op{gemm_based_op->get_operator()}, inputs, {mm});
+    }
+};
+
+MIGRAPHX_PRED_MATCHER(is_mlir_conv, instruction_ref ins)
+{
+    if(ins->name() != "convolution")
+        return false;
+    value v    = ins->get_operator().to_value();
+    auto group = v.at("group").to<int>();
+    if(group != 1)
+        return false;
+    // Avoid MLIR assertion: Index < Length && "Invalid index!"
+    if(ins->get_shape().lens().size() != 4)
+        return false;
+    return true;
+}
+
+struct find_conv_pointwise : public find_gemm_based_op_base<mlir_conv>
+{
+    // Find a convolution followed by a pointwise operation.
+    auto matcher() const
+    {
+        auto convolution =
+            match::skip(match::name("contiguous"))(is_mlir_conv().bind("gemm_based_op"));
+        return match::name("pointwise")(match::any_of[match::inputs()](convolution.bind("x")));
     }
 };
+
+MIGRAPHX_PRED_MATCHER(is_mlir_dot, instruction_ref ins)
+{
+    if(ins->name() != "dot")
+    {
+        return false;
+    }
+    return true;
+}
+
+struct find_dot_pointwise : public find_gemm_based_op_base<mlir_dot>
+{
+    // Find a dot followed by a pointwise operation.
+    auto matcher() const
+    {
+        auto dot = match::skip(match::name("contiguous"))(is_mlir_dot().bind("gemm_based_op"));
+        return match::name("pointwise")(match::any_of[match::inputs()](dot.bind("x")));
+    }
```

Comment:
These two matchers can be combined using `match::any_of(match::name("dot"), is_mlir_conv())`:

```cpp
auto matcher() const
{
  auto dot_or_conv = match::skip(match::name("contiguous"))(
      match::any_of(match::name("dot"), is_mlir_conv()).bind("gemm_based_op"));
  return match::name("pointwise")(
      match::any_of[match::inputs()](dot_or_conv.bind("x")));
}
```

Reply:
done